### PR TITLE
terminating app when error occurs

### PR DESCRIPTION
**Pull request by rybmat**

### CODE REVIEW

File: workers/common/src/main/java/com/wikia/workers/common/AbstractWorker.java [hunk 1]
```diff
@@ -82,11 +82,11 @@ public void run() {
         lastProcessed.update();
       } catch (Exception e) {
         LOGGER.error(e.getMessage(), e);
-        rabbitConnection.closeConnection();
-        System.exit(0);
+        break;
```
Comment (SebastianMarzjan) [inline workers/common/src/main/java/com/wikia/workers/common/AbstractWorker.java]: could stop() be called here instead to terminate the loop?

**Comment by idradm** [comment]
how about introducing shutdown hooks for connections?

http://stackoverflow.com/questions/3715967/when-should-we-call-system-exit-in-java
http://docs.oracle.com/javase/1.5.0/docs/guide/lang/hook-design.html

File: workers/common/src/main/java/com/wikia/workers/common/AbstractWorker.java [hunk 2]
```diff
@@ -65,6 +65,7 @@ public void stop() {
   }
 
   public void run() {
+    this.shutdownHooks();
```
Comment (idradm) [inline workers/common/src/main/java/com/wikia/workers/common/AbstractWorker.java]: Can we change this name to be more appropriate to what it does?

File: workers/common/src/main/java/com/wikia/workers/common/AbstractWorker.java [hunk 3]
```diff
@@ -104,4 +104,15 @@ protected void route(QueueingConsumer.Delivery delivery) throws Exception {
...
     throw new NotImplementedException("this method should be implemented in subclass");
   }
+
+  /**
+   * when overriding, remember to call super
+   */
+  protected void registerShutdownHooks() {
+    Runtime.getRuntime().addShutdownHook(new Thread() {
+      public void run() {
+        rabbitConnection.closeConnection();
```
Comment (idradm) [inline workers/common/src/main/java/com/wikia/workers/common/AbstractWorker.java]: "this" ?

Comment (rybmat) [inline workers/common/src/main/java/com/wikia/workers/common/AbstractWorker.java]: can not use it

**Comment by SebastianMarzjan** [comment]
Please resolve the conflicts; the current code looks good.

**Comment by SebastianMarzjan** [comment]
Once checks pass LGTM

**Pull request merged**